### PR TITLE
Refactor logging architecture

### PR DESCRIPTION
**Pull request by harshil21**
Changes:

- LDP is now a TypedDict, because we were instantly destroying a class right after instantiation
- Truncation now happens in logger process
- Less code

Bug fix:

- Predicted apogee is no longer logged with RawDataPackets

Performance:

Time to run test_integration.py:
On main: 19.42 seconds
This branch: 17.37 seconds

=> 10% faster

### CODE REVIEW

File: airbrakes/data_handling/logger.py [hunk 1]
```diff
@@ -158,61 +160,89 @@ def _log_the_buffer(self):
...
+
+                # Add the predicted apogee field:
+                logged_fields["predicted_apogee"] = predicted_apogee
+
+            logged_data_packets.append(logged_fields)
 
-            logged_data_packets.append(logged_data_packet)
         return logged_data_packets
+
+    # ------------------------------- RUN IN A SEPARATE PROCESS -----------------------------------
```
[the remark this answers is not in the document]
Comment (harshil21) [reply airbrakes/data_handling/logger.py:220]: it helps newcomers realize which methods run in a separate process

File: airbrakes/data_handling/logger.py [hunk 2]
```diff
@@ -54,14 +55,16 @@ def __init__(self, log_dir: Path) -> None:
...
-            writer = csv.DictWriter(file_writer, fieldnames=LoggedDataPacket.__struct_fields__)
+            writer = csv.DictWriter(file_writer, fieldnames=list(LoggedDataPacket.__annotations__))
             writer.writeheader()
 
         # Makes a queue to store log messages, basically it's a process-safe list that you add to
         # the back and pop from front, meaning that things will be logged in the order they were
         # added.
         # Signals (like stop) are sent as strings, but data is sent as dictionaries
-        self._log_queue: multiprocessing.Queue[dict[str, str] | str] = multiprocessing.Queue()
+        self._log_queue: multiprocessing.Queue[LoggedDataPacket | Literal["STOP"]] = (
```
[the remark this answers is not in the document]
Comment (harshil21) [reply airbrakes/data_handling/logger.py:65]: not in a `Literal` type. It has to be an Enum, or a actual string like that.

File: airbrakes/data_handling/logger.py [hunk 3]
```diff
@@ -158,61 +160,89 @@ def _log_the_buffer(self):
             self._log_queue.put(buffered_message)
         self._log_buffer.clear()
 
-    def _logging_loop(self) -> None:
+    def _truncate_decimal_place(self, obj_type: Any) -> str:
```
[the remark this answers is not in the document]
Comment (harshil21) [reply airbrakes/data_handling/logger.py:163]: I think methods which will only be used by the logger should not be in utils. 

But yeah I'm changing the name

File: airbrakes/data_handling/logger.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JacksonElia) [reply airbrakes/data_handling/logger.py:223]: add this every other proceess then

**Review by JacksonElia — APPROVED** [review]

**Pull request merged**